STACpopulator/input.py

```diff
@@ -23,18 +24,13 @@ def __init__(self) -> None:
         self.links = []
 
     @abstractmethod
-    def __iter__(self) -> Iterator:
+    def __iter__(self) -> Iterator[Tuple[str, str, MutableMapping[str, Any]]]:
         """
         Iterate over this loader.
 
-        Returns items from the input. Items may be anything depending on the specific concrete
-        implementation of this abstract class.
-        """
-        raise NotImplementedError
+        Returns items from the input. Items return a tuple containing the object name,
 
-    @abstractmethod
-    def reset(self) -> None:
```

Comment:
A reset method is no longer needed since the internal state of the iterator is no longer being saved as an instance variable on the object.